symmetry-core/src/main/java/org/biojava/nbio/structure/align/symm/CESymmCalculator.java

```diff
@@ -444,66 +453,56 @@ else if (score2 > userRMSDMax)
 	 * @param mse2
 	 * @param ca1
 	 * @param ca2
+	 * @param winSize
 	 * @return
 	 */
-	private double checkAngle(int mse1, int mse2, Atom[] ca1, Atom[] ca2,
+	private double checkAngle(int mse1, int mse2, Point3d[] ca1, Point3d[] ca2,
 			int winSize) {
 		try {
-
-			if (origin1 == null)
-				origin1 = Calc.getCentroid(ca1);
-			if (origin2 == null)
-				origin2 = Calc.getCentroid(ca2);
-
-			int nse1 = ca1.length;
-			int nse2 = ca2.length;
-
-			int max1 = Math.min((nse1 - mse1 - 1), winSize);
-			int max2 = Math.min((nse2 - mse2 - 1), winSize);
-
+			// length of next window
+			// TODO Doesn't the -1 skip the last residue? -Spencer 2018-12-27
+			int max1 = Math.min(ca1.length - mse1 - 1, winSize);
+			int max2 = Math.min(ca2.length - mse2 - 1, winSize);
 			int maxAtoms = Math.min(max1, max2);
 
-			// System.out.println(max1 + "  " + max2 + " " + maxAtoms);
-			Atom[] cod1 = new Atom[maxAtoms];
-			Atom[] cod2 = new Atom[maxAtoms];
-
-			assert (cod1.length == cod2.length);
-
-			// this sums up over the distances of the fragments
-
-			int is1 = mse1;
-			for (int pos1 = 0; pos1 < maxAtoms; pos1++) {
-				is1++;
-				cod1[pos1] = (Atom) ca1[is1].clone();
-				// System.out.println(pos1 + " " + is1 + " " + cod1[pos1]);
+			// Extract fragments
+ 			Point3d[] cod1 = new Point3d[maxAtoms];
+			Point3d[] cod2 = new Point3d[maxAtoms];
+			for(int i=0;i<maxAtoms;i++) {
+				cod1[i] = ca1[mse1+i];
+				cod2[i] = ca2[mse2+i];
 			}
-			int is2 = mse2;
-			for (int pos2 = 0; pos2 < maxAtoms; pos2++) {
-				is2++;
-				cod2[pos2] = (Atom) ca2[is2].clone();
-			}
-
-			SVDSuperimposer svd = new SVDSuperimposer(cod1, cod2);
 
-			Matrix matrix = svd.getRotation();
-			Atom shift = svd.getTranslation();
-
-			for (Atom a : cod2) {
-				Calc.rotate(a, matrix);
-				Calc.shift(a, shift);
-			}
-
-			Atom v1 = Calc.subtract(origin1, cod1[0]);
-			Atom v2 = Calc.subtract(origin2, cod2[0]);
-
-			double ang = Calc.angle(v1, v2);
-			// System.out.println(v1 + " " + v2 + " " + ang);
+			Matrix4d transform = SuperPositions.superpose(cod1, cod2);
+			
+			double ang = getAngle(transform);
+			
 			return ang;
 
 		} catch (Exception e) {
 			e.printStackTrace();
 			return 0d;
 		}
 	}
+	
+
+	/**
+	 * Quickly compute the rotation angle from a rotation matrix.
+	 * @param transform 4D transformation matrix. Translation components are ignored.
+	 * @return Angle, from 0 to PI
+	 * @deprecated Use {@link RotationAxis#rotationAngle} from biojava 5.1.2 onward
+	 */
+	@Deprecated
```

Comment:
Judging from the diff, this seems to be a new method. So there's no point in deprecating it, right? Or am I missing something? Since there seems to be an alternative in RotationAxis, I'd rather remove this one.